java/client/src/main/java/glide/api/commands/ListBaseCommands.java

```diff
@@ -501,4 +502,33 @@ CompletableFuture<Map<String, String[]>> blmpop(
      * }</pre>
      */
     CompletableFuture<String> lset(String key, long index, String element);
+
+    /**
+     * Atomically pops and removes the first/last element of the list stored at <code>source</code>
+     * depending on <code>wherefrom</code>, and pushes the element at the first/last element of the
+     * list stored at <code>destination</code> depending on <code>wherefrom</code>.
+     *
+     * @since Redis 6.2.0 and above.
+     * @apiNote When in cluster mode, <code>source</code> and <code>destination</code> must map to the
+     *     same hash slot.
+     * @see <a href="https://valkey.io/commands/lmove/">valkey.io</a> for details.
+     * @param source The key to the source list.
+     * @param destination The key to the destination list.
+     * @param wherefrom The direction the element should be removed from.
+     * @param whereto The direction the element should be added to.
+     * @return The popped element or null.
```

Comment:
```suggestion
     * @return The popped element or <code>null</code>.
```

Reply:
> ... or null

in which case?

Reply:
when `source` doesnt exist it returns null, should that be included in the javadoc?
> If source does not exist, the value nil is returned and no operation is performed.
https://redis.io/docs/latest/commands/lmove/

Reply:
Yes please